cmd/sonobuoy/app/gen.go

```diff
@@ -104,7 +104,7 @@ func (g *genFlags) Config() (*client.GenConfig, error) {
 
 	// the Enabled and Disabled modes of rbacmode don't need the client, so kubeclient can be nil.
 	// if kubeclient is needed, ErrRBACNoClient will be returned and that error can be reported back up.
-	rbacEnabled, err := genflags.rbacMode.Enabled(kubeclient)
+	rbacEnabled, err := g.rbacMode.Enabled(kubeclient)
```

Comment:
Tiny, mean bug that I found while coding these changes. It is referencing a global variable instead of the value it it was invoked on. This is why it happens to work for actual runs of Sonobuoy (since the global value gets used/set) but not in the test.

Very frustrating to find and another reason why we shouldn't be relying on them. One day we'll get around to fixing all the globals for these cmds...one day...